core/src/consensus/tendermint/worker.rs

```diff
@@ -1708,12 +1708,17 @@ impl Worker {
                 // NOTE: Only the genesis block and the snapshot target don't have the parent in the blockchain
                 None => continue,
             };
-            let term_seconds = if let Some(p) = c.term_common_params(parent_header.hash().into()) {
+            let grandparent_header = match c.block_header(&BlockId::Hash(*header.parent_hash())) {
+                Some(h) => h.decode(),
+                // Grandparent can be missing if the block number < 2 or there is a missing
+                None => continue,
+            };
+            let term_seconds = if let Some(p) = c.term_common_params(grandparent_header.hash().into()) {
                 p.term_seconds()
             } else {
                 continue
             };
-            if super::engine::is_term_changed(&header, &parent_header, term_seconds) {
+            if super::engine::is_term_changed(&parent_header, &grandparent_header, term_seconds) {
```

Comment:
How about using this instead?
```
header.number() == parent.last_term_finished_block_num() + 1
```
It's used by `on_open_block` to check if it's the term's starting block.

Reply:
Resolved